### ISSUE TITLE
Encoding issue when xml contains : "<x>&gt;10</x>"

### ISSUE THREAD

**Issue by edi9999** [xmldom/xmldom#22]
See the following example with xmldom 0.2.1

```
var DOMParser = require('xmldom').DOMParser;
var XMLSerializer = require('xmldom').XMLSerializer;

const str = "<x>&gt;10</x>"
var doc = new DOMParser().parseFromString(str ,'text/xml');
var newStr = new XMLSerializer().serializeToString(doc);

console.log(str);
console.log(newStr);
```

If I run this in the browser without xmldom on firefox and chrome , I get : 

```
<x>&gt;10</x>
<x>&gt;10</x>
```

however, in xmldom, the string is replaced wrongfully to `>` : 

```
<x>&gt;10</x>
<x>>10</x>
```

**Comment by edi9999** [comment]
Note the issue is present for `&gt;`, but not for `&lt;` neither for `&amp;`

**Comment by DiegoMajluf** [comment]
I have the same problem but the issue is present for &gt; &quot; and &apos;

**Comment by DiegoMajluf** [comment]
I have the same problema but the issue is present not only for `&gt;` but also for `&quot;` and `&apos;`

**Comment by DiegoMajluf** [comment]
I checked and it's seems that xmldom serialization, even that is different to XMLSerializer is a legitime XML document any way

**Comment by edi9999** [comment]
Still, it would be cleaner imo to have the same behavior.

**Comment by microshine** [comment]
I've got the same problem. Different results for XML node with namespace and without

```js
const items = [
  `<html xmlns="http://www.w3.org/1999/xhtml"><script>let message = " &amp; ETH";</script></html>`,
  `<html><script>let message = " &amp; ETH";</script></html>`,
];

for (const item of items) {
  const xml = new DOMParser().parseFromString(item, "application/xml");
  console.log(xml.documentElement.childNodes[0].textContent);
}
```

### NodeJS  output
```
let message = " &amp; ETH";
let message = " & ETH";
```

### Browser output
Using native DOMParaer class
```
let message = " & ETH";
let message = " & ETH";
```

**Comment by karfau** [comment]
See/Subscribe to #58 for further updates on this topic.

**Comment by colourful-land** [comment]
> See/Subscribe to #58 for further updates on this topic.

This issue is not solved by the resolution of #58 

First, #58 deals with `]]>` which isn't mentioned in this issue. This one is about `&amp;`

Then, with the patched-up xmldom master this issue is still reproducible.

````
$ cat xmldom_bug_22.js 
const { DOMParser } = require('xmldom')

const items = [
  `<html xmlns="http://www.w3.org/1999/xhtml"><script>let message = " &amp; ETH";</script></html>`,
  `<html><script>let message = " &amp; ETH";</script></html>`,
];

for (const item of items) {
  const xml = new DOMParser().parseFromString(item, "application/xml");
  console.log(xml.documentElement.childNodes[0].textContent);
}
$ node xmldom_bug_22.js 
let message = " &amp; ETH";
let message = " & ETH";
````

Where both messages should be the same.
